### ISSUE TITLE
[type, number -> int|float] Makes no sense, because this type is not defined in annotations

### ISSUE THREAD

**Issue by BenMorel** [kalessil/phpinspectionsea#516]
I have the following function:

```php
/**
 * @param number|string $amount
 */
function add($amount)
{
    if (is_int($amount)) {
        
    }
}
```

The `is_int()` function is highlighted with the following warning:

> Makes no sense, because this type is not defined in annotations.

Actually, the `int` type *is* defined in the annotation, but implicitly: `number` is a synonym for `int|float`!

**Comment by kalessil** [comment]
Hm, https://docs.phpdoc.org/guides/types.html#primitives doesn't have type `number`.
Let me check how often the type is used in OSS project - if so I'll add the type translation or at least mapping to `int` if restricted by our APIs.

Thanks for hands on!

**Comment by BenMorel** [comment]
PHPDoc is often cited as the de-facto standard, but the fact is, there is no such standard yet. I'm personally using `number` in my [docs](https://github.com/brick/math/blob/d1f3fdc884cf012567c0a6f300c1c10cb8b2d64c/src/BigDecimal.php#L136) as a shortcut for `int|float`, and my IDE, PHPStorm, does seem to understand it.

This is subject to debate, though, I can appreciate that.

**Comment by rentalhost** [comment]
Please, vote here: https://youtrack.jetbrains.com/issue/WI-37804

**Comment by kalessil** [comment]
Voted as well, added support on our side.

**Comment by BenMorel** [comment]
Thanks for the support! I'm still having the same warning with 2.3.10.2, though?

**Comment by BenMorel** [comment]
Ah sorry, just realized that the commit happened *after* the release. Will wait for the next one then ;-)

**Comment by kalessil** [comment]
The next release =) Or if I discovered/get reported new java exceptions, then in 2.3.10.3.

**Comment by kalessil** [comment]
@BenMorel : it'll be 2.3.10.3 tomorrow evening, so the change will be there =)

**Comment by BenMorel** [comment]
I can confirm it works perfectly now, thanks!
